### PR TITLE
Create Signup/Login API Endpoints

### PR DESCRIPTION
**Pull request by SilverStrings024**
Created Signup/Login/Logout api endpoints.
Created authenticate decorator to ensure the user is logged in and has a valid token
Created refresh function for refreshing a JWT
Also updated app.py to configure flask_extended_jwt
Tested with Postman so everything should work
**NOTE** - This relies on User.get_fields() but that hasn't been committed yet and GitHub Desktop is preventing me from changing branches to do so. It will be in a PR as soon as I get that figured out

### CODE REVIEW

File: server/api/auth.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        user_info = {}
+        for field in User.get_fields():
+            if field == "id":
+                continue
+            if field == "password_hash":
+                user_info['password'] = request_dict['password']
+                continue
+            user_info[field] = request_dict[field]
+        try:
+            user = User.create(**user_info)
```
Comment (sina-jamshidi) [inline server/api/auth.py:64]: this will automatically let us know if the user already exists right?

File: server/api/auth.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    make_response, redirect,
+    request, session,
+    url_for, g
+)
+from flask_jwt_extended import (
+    JWTManager, jwt_required, create_access_token,
+    jwt_refresh_token_required, create_refresh_token,
+    get_jwt_identity, set_access_cookies,
+    set_refresh_cookies, unset_jwt_cookies,
+    jwt_optional, verify_jwt_in_request
```
Comment (sina-jamshidi) [inline server/api/auth.py]: it looks like you're not using jwt_optional

File: server/api/auth.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
+import functools
+import json
+from sqlalchemy.orm.exc import NoResultFound
+from flask import Blueprint as bp
+from flask import (
+    make_response, redirect,
+    request, session,
+    url_for, g
```
Comment (sina-jamshidi) [inline server/api/auth.py]: looks like you're not using redirect and url_for

**Review by sina-jamshidi — APPROVED** [review]
Looks good! Just remove a couple of unused things and good to merge

File: server/api/auth.py (hunk 1 above)
Comment (SilverStrings024) [reply server/api/auth.py:64]: Yes it has a method in it to check if a user with the given username or email exists already.

File: server/api/auth.py (hunk 2 above)
Comment (SilverStrings024) [reply server/api/auth.py]: Good eye, thanks!

File: server/api/auth.py (hunk 3 above)
Comment (SilverStrings024) [reply server/api/auth.py]: Alright, I'll change that.

**Comment by SilverStrings024** [comment]
> Looks good! Just remove a couple of unused things and good to merge

Thanks! I'll put up a PR tonight or tomorrow morning, fighting with postgres with a new seeding method to populate the database if we have to drop it.

**Pull request merged**